From 1f424ad911ce4f704885ef5682ea007300624dbe Mon Sep 17 00:00:00 2001
From: Zong Li <zong.li@sifive.com>
Date: Thu, 11 Mar 2021 17:23:22 +0800
Subject: [PATCH 50/52] riscv: sifive: fu740: Support i2c in spl

Enable SPL_I2C_SUPPORT for fu740, and add 'u-boot,dm-spl' property in
i2c node as well.

Signed-off-by: Zong Li <zong.li@sifive.com>
---
 arch/riscv/cpu/fu740/Kconfig          | 1 +
 arch/riscv/dts/fu740-c000-u-boot.dtsi | 4 ++++
 2 files changed, 5 insertions(+)

diff --git a/arch/riscv/cpu/fu740/Kconfig b/arch/riscv/cpu/fu740/Kconfig
index 0f6ffe28d5..590fd53a81 100644
--- a/arch/riscv/cpu/fu740/Kconfig
+++ b/arch/riscv/cpu/fu740/Kconfig
@@ -37,3 +37,4 @@ config SIFIVE_FU740
 	imply PWM_SIFIVE
 	imply DM_I2C
 	imply SYS_I2C_OCORES
+	imply SPL_I2C_SUPPORT
diff --git a/arch/riscv/dts/fu740-c000-u-boot.dtsi b/arch/riscv/dts/fu740-c000-u-boot.dtsi
index b6f7545c40..7aba797042 100644
--- a/arch/riscv/dts/fu740-c000-u-boot.dtsi
+++ b/arch/riscv/dts/fu740-c000-u-boot.dtsi
@@ -95,6 +95,10 @@
 	u-boot,dm-spl;
 };
 
+&i2c0 {
+	u-boot,dm-spl;
+};
+
 &eth0 {
 	assigned-clocks = <&prci PRCI_CLK_GEMGXLPLL>;
 	assigned-clock-rates = <125125000>;
-- 
2.30.2

